### ISSUE TITLE
Improve save() method in torch.jit.ScriptModule

### ISSUE THREAD

**Issue by laughmetal** [pytorch/pytorch#17688]
## 🚀 Feature
save() method in torch.jit.ScriptModule should support save_to_buffer as well.

## Motivation
Right now, to save a torch.jit.ScriptModule to buffer, I have to call function torch.jit.save().
It would be more convenient and consistent if torch.jit.ScriptModule.save() can perform the same job.

## Pitch
I can work on this feature.

**Comment by eellison** [comment]
Thanks for the feature request! We will discuss this and get back to you

**Comment by suo** [comment]
This would be decently easy, since it's already in the c++ module type. @laughmetal if you'd like to work on it, we would be happy to review a PR! 😄

**Comment by laughmetal** [comment]
You are right. It's probably just a [binding](https://github.com/pytorch/pytorch/blob/a2381fa346a08dd9be3ef7942efb048d789cbee1/torch/csrc/jit/script/init.cpp#L689) decision. Should we change the binging, or should we add a new session to the doc after [this](https://pytorch.org/docs/stable/jit.html#torch.jit.ScriptModule.save)?

**Comment by suo** [comment]
I think it would be reasonable to make ScriptModule.save() and torch.jit.save() consistent. So ScriptModule.save() would either save to a buffer or a file depending on the passed argument.

**Comment by WilliamTambellini** [comment]
@laughmetal  I supposed this is not yet implemented:
https://pytorch.org/docs/master/generated/torch.jit.ScriptModule.html#torch.jit.ScriptModule.save
?

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
